core/src/main/java/org/apache/carbondata/core/index/CarbonIndexProvider.java

```diff
@@ -72,91 +72,95 @@
  * @since 1.4.0
  */
 @InterfaceAudience.Internal
-public abstract class DataMapProvider {
+public abstract class CarbonIndexProvider {
```

Comment:
```suggestion
public abstract class IndexProvider {
```

Reply:
Already we have a class with same name. Cannot rename with same